src/app/shared/equation.ts

```diff
@@ -37,6 +41,24 @@ class Constant implements Term {
   getValue(): ScalarAndDimension { return this.value; }
 
   contains(v: Variable): boolean { return false; }
+
+  log(base: Term, antilogarithm: Term): ScalarAndDimension | CalculateErrors | null {
+    const baseValue = base.getValue();
+    if (baseValue == null || isCalculateError(baseValue)) {
+      return baseValue;
+    }
+    const logValue = Math.log(this.value.n) / Math.log(baseValue.n);
+    return new ScalarAndDimension(logValue, this.value.d);
+  }
+
+  antilog(base: Term, logarithm: Term): ScalarAndDimension | CalculateErrors | null {
+    const baseValue = base.getValue();
+    if (baseValue == null || isCalculateError(baseValue)) {
+      return baseValue;
+    }
+    const antilogValue = Math.pow(baseValue.n, this.value.n);
+    return new ScalarAndDimension(antilogValue, this.value.d);
+  }
```

Comment:
log and antilog don't belong on Constant. For instance, add and mul aren't here. Instead, they need to have their own classes which extend Term, just like Add and Multiply do. Note that Logarithmize class is already there but stubbed out, while Antilogarithmize needs to be added.

Then, methods which return them, static methods on Equation should be added, just like add, sub, mul and div.